swift/aperture/main.swift

```diff
@@ -2,37 +2,51 @@ import Foundation
 import AVFoundation
 
 var recorder: Recorder!
+let arguments = CommandLine.arguments.dropFirst()
 
 func quit(_: Int32) {
   recorder.stop()
 }
 
-func record() throws {
-  // TODO: Use JSON and `Codable` here when Swift 4 is out
+struct CropArea: Decodable {
+  let x: CGFloat
+  let y: CGFloat
+  let width: CGFloat
+  let height: CGFloat
+}
+
+struct Options: Decodable {
+  let destination: String // TODO: Figure out a way to make this decodable into an `URL`
```

Comment:
I thought that too, but when I tried `URL` there I got an error:

> Cannot record to URL /private/var/folders/3x/jf5977fn79jbglr7rk0tq4d00000gn/T/32e3df1e53d027351c7edbfe9db2a72f.mp4 because it is not a file URL\'\n

So it seems it will only correctly coerce if you specify a file URL. I could do that on the JS side, but it feels like something I should be able to handle with a custom decodable handler or something.

Reply:
Add file:// infront